src/nestjs-back/src/chat/channels/channels.service.ts

```diff
@@ -107,17 +108,17 @@ export class ChannelsService {
   }
 
   async update(id: string, updateChannelDto: UpdateChannelDto) {
-    let channel = await this.channelsRepository.preload({
+    if (updateChannelDto.password) {
+      checkPasswordPattern(updateChannelDto.password);
```

Comment:
I think we should be able to use a custom validator for the password pattern instead of doing the check inside the service
see https://github.com/typestack/class-validator#custom-validation-decorators